### PR TITLE
[DTA-1034] Integrate BugSnag error monitoring in the Payment Service

### PR DESCRIPTION
**Pull request by AdolfoLopezSB**
# Changes

Installed Bugnsag Error Monitoring into the payment service.
[DTA-1034](https://smartbear.atlassian.net/browse/DTA-1034)

[DTA-1034]: https://smartbear.atlassian.net/browse/DTA-1034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/paymentservice/charge.js [hunk 1]
```diff
@@ -12,12 +12,14 @@ const logger = require('./logger');
...
 const meter = metrics.getMeter('paymentservice');
 const transactionsCounter = meter.createCounter('app.payment.transactions')
+const Bugsnag = require("@bugsnag/js");
 
 module.exports.charge = async request => {
   const span = tracer.startSpan('charge');
 
   await OpenFeature.setProviderAndWait(flagProvider);
   if (await OpenFeature.getClient().getBooleanValue("paymentServiceFailure", false)) {
+    Bugsnag.notify(new Error("PaymentService Fail Feature Flag Enabled"));
```
Comment (LucjanMichalowski) [inline src/paymentservice/charge.js]: Bugsnag by default should report unhandled errors. If you remove this line, does it still send an error?

Comment (AdolfoLopezSB) [reply src/paymentservice/charge.js]: I couldn't manage to make it work without it. But having a look into the documentation, I understood that only unhandled exceptions and unhandled promises would be reported, so I assumed that it's the expected behaviour.
image

**Review by LucjanMichalowski — COMMENTED** [review]
LGTM

A note, that before merging and deploying it, we need to add the `PAYMENT_SERVICE_BUGSNAG_API_KEY` to GitHub secrets.

**Review by mottibec — APPROVED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -439,6 +439,7 @@ services:
       - "${PAYMENT_SERVICE_PORT}"
     environment:
       - PAYMENT_SERVICE_PORT
+      - PAYMENT_SERVICE_BUGSNAG_API_KEY
```
Comment (LucjanMichalowski) [inline docker-compose.yml]: Based on how the Terraform configuration looks, it seems that we need to keep the `BUGSNAG_API_KEY` in the app and add the mapping in here, so what should work is:
```
- BUGSNAG_API_KEY=${PAYMENT_SERVICE_BUGSNAG_API_KEY}
```
So in the `.env` file we will have `PAYMENT_SERVICE_BUGSNAG_API_KEY` and in the code in the app `BUGSNAG_API_KEY`.

Please let me know if it works.

Comment (AdolfoLopezSB) [reply docker-compose.yml]: I made the change and it works, I could still see the error in Bugnsag

**Review by LucjanMichalowski — APPROVED** [review]

**Pull request merged**